### PR TITLE
feat: use new generics format

### PR DESCRIPTION
**Pull request by vishalnarkhede**
## 🎯 Goal

This PR focuses on converting the Stream chat generics into a single generics making the code simpler.

## 🛠 Implementation details

The previously written generics are now converted to a single generic as an object of the different types(attachment, channel, etc.). 

## 🎨 UI Changes

No such UI change is done in this PR.

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [x] New code is covered by tests
- [ ] Screenshots added for visual changes
- [x] Documentation is updated

### CODE REVIEW

**Review by madsroskar — APPROVED** [review]

**Comment by madsroskar** [comment]
lgtm, but we could do a pass through the documentation with Grammarly or similar, I think

File: docusaurus/docs/reactnative/customization/typescript.mdx [hunk 1]
```diff
@@ -92,7 +121,7 @@ The [TypeScript Example App](https://github.com/GetStream/stream-chat-react-nati
 Core to understanding this usage is [how generics can be applied to JSX elements](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#generic-type-arguments-in-jsx-elements).
 
 In many cases the use of a single prop such as `client` or `channel` allows TypeScript to infer the generics on an element.
-In this case `LocalAttachmentType` is inferred from `channel` and passed to the props type for a custom Attachment component.
+In this case `StreamChatType` is inferred from `channel` and passed to the props type for a custom Attachment component.
```
[the remark this answers is not in the document]
Comment (vishalnarkhede) [reply docusaurus/docs/reactnative/customization/typescript.mdx:124]: Indeed you are correct

File: docusaurus/docs/reactnative/customization/typescript.mdx [hunk 2]
```diff
@@ -21,20 +21,48 @@ In many cases TypeScript can use [inference](https://www.typescriptlang.org/docs
...
+  commandType: LocalCommandType;
+  eventType: LocalEventType;
+  messageType: LocalMessageType;
+  reactionType: LocalReactionType;
+  userType: LocalUserType;
+};
+```
+
+:::note
+`LocalAttachmentType`, `LocalChannelType` etc. are type deifinitions for their respective key as per your application types necessities.
```
Comment (vishalnarkhede) [inline docusaurus/docs/reactnative/customization/typescript.mdx:45]:
```suggestion
`LocalAttachmentType`, `LocalChannelType` etc. are type definitions for their respective key as per your application types necessities.
```

File: docusaurus/docs/reactnative/customization/typescript.mdx (hunk 1 above)
Comment (vishalnarkhede) [inline docusaurus/docs/reactnative/customization/typescript.mdx:124]:
```suggestion
In this case `channelType` within `StreamChatType` is inferred from `channel` and passed to the props type for a custom Attachment component.
```

File: docusaurus/docs/reactnative/customization/typescript.mdx [hunk 3]
```diff
@@ -64,19 +89,23 @@ type DefaultUserType = Record<string, unknown> & {
 };
 ```
 
-Additional fields on the defaults i.e. `file_size`, `mime_type`, and `image` are custom fields used by `stream-chat-react-native` already within the SDK.
+Additional fields on the defaults i.e. `image` are custom fields used by `stream-chat-react-native` already within the SDK.
```
Comment (vishalnarkhede) [inline docusaurus/docs/reactnative/customization/typescript.mdx]:
```suggestion
Additional fields on the defaults i.e. `image` is a custom fields used by `stream-chat-react-native` already within the SDK.
```

**Pull request merged**